connectors/sources/google_drive.py

```diff
@@ -0,0 +1,638 @@
+    "application/vnd.google-apps.spreadsheet": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+}
+
+GOOGLE_DRIVE_EMULATOR_HOST = os.environ.get("GOOGLE_DRIVE_EMULATOR_HOST")
+RUNNING_FTEST = (
+    "RUNNING_FTEST" in os.environ
+)  # Flag to check if a connector is run for ftest or not.
+DEFAULT_PEM_FILE = os.path.join(
+    os.path.dirname(__file__),
+    "..",
+    "..",
+    "tests",
+    "sources",
+    "fixtures",
+    "google_drive",
+    "service_account_dummy_cert.pem",
+)
+
+
+class GoogleDriveClient:
+    """A google client to handle api calls made to Google Drive."""
+
+    def __init__(self, retry_count, json_credentials):
+        """Initialize the ServiceAccountCreds class using which api calls will be made.
+
+        Args:
+            retry_count (int): Maximum retries for the failed requests.
+            json_credentials (dict): Service account credentials json.
+        """
+        self.retry_count = retry_count
+        self.service_account_credentials = ServiceAccountCreds(
+            scopes=[GOOGLE_DRIVE_READ_ONLY_SCOPE],
+            **json_credentials,
+        )
+
+    async def api_call(
+        self,
+        resource,
+        method,
+        sub_method=None,
+        full_response=False,
+        **kwargs,
+    ):
+        """Make a GET call to Google Drive API with retry for the failed calls.
+
+        Args:
+            resource (aiogoogle.resource.Resource): Resource name for which api call will be made.
+            method (aiogoogle.resource.Method): Method available for the resource.
+            sub_method (aiogoogle.resource.Method, optional): Sub-method available for the method. Defaults to None.
+            full_response (bool, optional): Specifies whether the response is paginated or not. Defaults to False.
+
+        Raises:
+            exception: A instance of an exception class.
+
+        Yields:
+            Dictionary: Response returned by the resource method.
+        """
+        retry_counter = 0
+        while True:
+            try:
+                async with Aiogoogle(
+                    service_account_creds=self.service_account_credentials
+                ) as google_client:
+                    drive_client = await google_client.discover(
+                        api_name=API_NAME, api_version=API_VERSION
+                    )
+                    if RUNNING_FTEST and not sub_method and GOOGLE_DRIVE_EMULATOR_HOST:
+                        logger.debug(
+                            f"Using the google drive emulator at {GOOGLE_DRIVE_EMULATOR_HOST}"
+                        )
+                        # Redirecting calls to fake Google Drive server for e2e test.
+                        drive_client.discovery_document["rootUrl"] = (
+                            GOOGLE_DRIVE_EMULATOR_HOST + "/"
+                        )
+                    resource_object = getattr(drive_client, resource)
+                    method_object = getattr(resource_object, method)
```

Comment:
because the for e.g. `files/list` endpoint we need to access `drive_client.files.list` method object. 

Since `resource` and `method` are api_call params this is the way to programatically access the object. Note that (resource, method) are different for different api calls, they can be:
- (`files`, `list`)
- (`files`, `get`)
- (`drives`, `list`)
- ...